internal/rename/controller/executor.go

```diff
@@ -0,0 +1,46 @@
+/*
+ *  *******************************************************************************
+ *  * Copyright (c) 2019 Edgeworx, Inc.
+ *  *
+ *  * This program and the accompanying materials are made available under the
+ *  * terms of the Eclipse Public License v. 2.0 which is available at
+ *  * http://www.eclipse.org/legal/epl-2.0
+ *  *
+ *  * SPDX-License-Identifier: EPL-2.0
+ *  *******************************************************************************
+ *
+ */
+
+package controller
+
+import (
+	"fmt"
+	"github.com/eclipse-iofog/iofogctl/internal/config"
+	"github.com/eclipse-iofog/iofogctl/pkg/util"
+)
+
+func Execute(namespace, name, newname string) error {
+	// Check that Controller exists in current namespace
+	_, err := config.GetController(namespace, name)
+	if err != nil {
+		return err
+	}
+
+	util.SpinStart(fmt.Sprintf("Renaming Controller %s", name))
+
+	// Do a shallow rename of controller
+	ns, err := config.GetNamespace(namespace)
+	if err != nil {
+		return err
+	}
+	for i, ctrl := range ns.ControlPlane.Controllers {
```

Comment:
As per Connector and Agent comments. use config.UpdateController() and do not loop over this to perform update